### PR TITLE
[Test] Create automated test for "Existing workspace found" scenario

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
We should test this scenario works to avoid regressions like [this one](https://github.com/eclipse/che/issues/22210):

### What issues does this PR fix or reference?
This PR is a part of https://issues.redhat.com/browse/CRW-4652 issue.
Testing steps:
1. Create a first workspace.
2. Create a second workspace with the same name
3. When warned that "Existing workspace found" click on "Create a new workspace"
4. Confirm that the new workspace is started

### How to test this PR?
That is possibly to start this test with any stack except EmptyWorkspace(it is always will be created with different name).

To start test set  next constants:
export USERSTORY=CreateWorkspaceWithExistedName
export TS_SELENIUM_DASHBOARD_SAMPLE_NAME=stack name(Python by default)
export TS_SELENIUM_PROJECT_NAME=project name(python-hello-world by default)

start `npm run test` command

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by nallikaea — APPROVED** [review]

File: tests/e2e/tests-library/WorkspaceHandlingTests.ts [hunk 1]
```diff
@@ -73,6 +73,13 @@ export class WorkspaceHandlingTests {
 		await this.browserTabsUtil.waitAndSwitchToAnotherWindow(WorkspaceHandlingTests.parentGUID, TIMEOUT_CONSTANTS.TS_IDE_LOAD_TIMEOUT);
 	}
 
+	async createAndOpenWorkspaceWithExistedWorkspaceName(stack: string): Promise<void> {
+		Logger.debug('create and open workspace with existed workspace name.');
```
Comment (dmytro-ndp) [inline tests/e2e/tests-library/WorkspaceHandlingTests.ts:77]: There is no need to enter method description like `'create and open workspace with existed workspace name.'` when call Logger methods after commit https://github.com/eclipse/che/commit/df80cc80dcf00321cf839278cd516aed5b6efc87

File: tests/e2e/specs/miscellaneous/CreateWorkspaceWithExistedName.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { WorkspaceHandlingTests } from '../../tests-library/WorkspaceHandlingTests';
+import { ProjectAndFileTests } from '../../tests-library/ProjectAndFileTests';
+import { LoginTests } from '../../tests-library/LoginTests';
+import { registerRunningWorkspace } from '../MochaHooks';
+import { BrowserTabsUtil } from '../../utils/BrowserTabsUtil';
+import { BASE_TEST_CONSTANTS } from '../../constants/BASE_TEST_CONSTANTS';
+import { Dashboard } from '../../pageobjects/dashboard/Dashboard';
+import { FACTORY_TEST_CONSTANTS } from '../../constants/FACTORY_TEST_CONSTANTS';
+
+const stackName: string = BASE_TEST_CONSTANTS.TS_SELENIUM_DASHBOARD_SAMPLE_NAME || 'Python';
```
Comment (dmytro-ndp) [inline tests/e2e/specs/miscellaneous/CreateWorkspaceWithExistedName.spec.ts:23]: Isn't it `sampleName` ?

**Comment by dmytro-ndp** [comment]
@SkorikSergey : do you have test run logs to review?

**Comment by SkorikSergey** [comment]
`[sskoryk@fedora e2e]$ npm run test

> @eclipse-che/che-e2e@7.73.0-SNAPSHOT test
> ./configs/sh-scripts/initDefaultValues.sh npm run lint && npm run tsc && export USERSTORY=$USERSTORY && mocha --config dist/configs/mocharc.js

Initialized default values

TS_SELENIUM_VALUE_TLS_SUPPORT =       
TS_SELENIUM_VALUE_OPENSHIFT_OAUTH =   true
TS_OCP_LOGIN_PAGE_PROVIDER_TITLE =    htpasswd
E2E_OCP_CLUSTER_VERSION =             4.x
TS_SELENIUM_LOG_LEVEL =               DEBUG
TS_SELENIUM_OCP_USERNAME =            admin
TS_SELENIUM_W3C_CHROME_OPTION =       true
NODE_TLS_REJECT_UNAUTHORIZED =        0
TS_SELENIUM_EDITOR =                  che-code

> @eclipse-che/che-e2e@7.73.0-SNAPSHOT tsc
> rm -rf ./dist && ./configs/sh-scripts/generateIndex.sh && tsc -p .

Generating index.ts file...
Warning: Cannot find any files matching pattern "dist/specs/CreateWorkspaceWithExistedName.spec.js"

################## Launch Information ##################

      TS_SELENIUM_BASE_URL: https://devspaces.apps.ocp412-sskoryk.crw-qe.com
      TS_SELENIUM_HEADLESS: false
      TS_SELENIUM_OCP_USERNAME: admin
      TS_SELENIUM_EDITOR:   che-code

      TS_SELENIUM_HAPPY_PATH_WORKSPACE_NAME: EmptyWorkspace
      TS_SELENIUM_DELAY_BETWEEN_SCREENSHOTS: 1000
      TS_SELENIUM_REPORT_FOLDER: ./report
      TS_SELENIUM_EXECUTION_SCREENCAST: false
      DELETE_SCREENCAST_IF_TEST_PASS: true
      TS_SELENIUM_REMOTE_DRIVER_URL: 
      DELETE_WORKSPACE_ON_FAILED_TEST: true
      TS_SELENIUM_LOG_LEVEL: INFO
      TS_SELENIUM_LAUNCH_FULLSCREEN: true

      TS_COMMON_DASHBOARD_WAIT_TIMEOUT: 5000
      TS_SELENIUM_START_WORKSPACE_TIMEOUT: 360000
      TS_WAIT_LOADER_PRESENCE_TIMEOUT: 60000

      TS_SAMPLE_LIST: Node.js MongoDB,Node.js Express

      MOCHA_DRIRECTORY is not set
      USERSTORY: CreateWorkspaceWithExistedName

      to output timeout variables, set TS_SELENIUM_PRINT_TIMEOUT_VARIABLES to true
 ######################################################## 


  "Start workspace with existed workspace name" test
    ✔ Login (12013ms)
(node:52163) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
      • WorkspaceHandlingTests.obtainWorkspaceNameFromStartingPage - obtained workspace name from workspace loader page: python-hello-world
    ✔ Create and open new workspace, stack:Python (6968ms)
    ✔ Wait workspace readiness and project folder has been created (18163ms)
    ✔ Stop created workspace (4705ms)
      • WorkspaceHandlingTests.obtainWorkspaceNameFromStartingPage - obtained workspace name from workspace loader page: python-hello-world-99cq
    ✔ Create new workspace from the same Python stack (10008ms)
    ✔ Wait the second workspace readiness and project folder has been created (25576ms)
    ✔ Stop all created Python workspaces (6790ms)
    ✔ Delete all created Python workspaces (16296ms)
    ✔ Logout (7332ms)

      • Context.stopTheDriver - Chrome driver session stopped.

  9 passing (2m)`

**Comment by dmytro-ndp** [comment]
@SkorikSergey : thank you.
Looks good, IMHO.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**